### DISCUSSION TITLE
read_multi_vars limitation of 20 tags at once, is at Siemens PLC level or at python snap7 level

### DISCUSSION THREAD

**Discussion by Rahul14189** [gijzelaerr/python-snap7#461] [Q&A] [answered]
Does 'read_multi_vars' method's limitation of reading 20 addresses (or tags) at once, is because of Siemens PLC or it is in python snap7 only !

**Comment by nikteliy** [comment, marked as answer]
It is a limitation of `snap7` library. 

At the `s7comm` protocol level, requests are limited by PDU length, so you need to calculate whether both the request data and the response data fit within the PDU. Additionally, requests should not be long due to the possibility of data changes during the request process.
The limitation on the maximum number of variables is a ~~somewhat lazy~~ elegant way to mitigate risks and simplify the code.
![snap7](https://github.com/gijzelaerr/python-snap7/assets/52915342/28cca5b9-c6bf-4353-9690-7d8474c2a645)

**Reply by Rahul14189** [reply]
Thanks
